Messaging.md

```diff
@@ -0,0 +1,63 @@
+# Messaging Protocol
+
+## Overview
+
+This protocol assumes an underlying authenticated and ordered transport mechanism that takes care of framing individual messages.
+[BOLT #8](08-transport.md) specifies the canonical transport layer used in Lightning, though it can be replaced by any transport that fulfills the above guarantees.
+
+All data fields are unsigned big-endian unless otherwise specified.
+
+## Table of Contents
+
+* [Connection Handling and Multiplexing](#connection-handling-and-multiplexing)
+  * [Message Format](#message-format)
+  * [Fundamental Types](#fundamental-types)
+* [Authors](#authors)
+
+## Connection Handling and Multiplexing
+
+Implementations MUST use a single connection per peer; contract messages (which include a contract ID) are multiplexed over this single connection.
+
+## Message Format
+
+We reuse the [Lightning Message Format](01-messaging.md#lightning-message-format) and the [Type-Length-Value Format](01-messaging.md#type-length-value-format)
+
+## Fundamental Types
+
+Various fundamental types are referred to in the message specifications:
+
+* `byte`: an 8-bit byte
+* `u16`: a 2 byte unsigned integer
+* `u32`: a 4 byte unsigned integer
+* `u64`: an 8 byte unsigned integer
+
+Inside TLV records which contain a single value, leading zeros in
+integers can be omitted:
+
+* `tu16`: a 0 to 2 byte unsigned integer
+* `tu32`: a 0 to 4 byte unsigned integer
+* `tu64`: a 0 to 8 byte unsigned integer
+
+The following convenience types are also defined:
+
+* `chain_hash`: a 32-byte chain identifier (see [BOLT #0](00-introduction.md#glossary-and-terminology-guide))
```

Comment:
link doesn't work use

https://github.com/lightningnetwork/lightning-rfc/blob/master/00-introduction.md#chain_hash

Reply:
Definition should be moved into this repo eventually however